crates/fj-export/src/lib.rs

```diff
@@ -16,33 +16,33 @@
 
 use std::{fs::File, path::Path};
 
-use anyhow::{anyhow, Result};
+use thiserror::Error;
 
 use fj_interop::mesh::Mesh;
-use fj_math::{Point, Triangle, Vector};
+use fj_math::{Point, Triangle};
 
 /// Export the provided mesh to the file at the given path.
 ///
 /// This function will create a file if it does not exist, and will truncate it if it does.
 ///
 /// Currently 3MF & STL file types are supported. The case insensitive file extension of
 /// the provided path is used to switch between supported types.
-pub fn export(mesh: &Mesh<Point<3>>, path: &Path) -> Result<()> {
+pub fn export(mesh: &Mesh<Point<3>>, path: &Path) -> Result<(), Error> {
     match path.extension() {
         Some(extension) if extension.to_ascii_uppercase() == "3MF" => {
             export_3mf(mesh, path)
         }
         Some(extension) if extension.to_ascii_uppercase() == "STL" => {
             export_stl(mesh, path)
         }
-        Some(extension) => {
-            Err(anyhow!("Extension not recognised, got {:?}", extension))
-        }
-        None => Err(anyhow!("No extension specified")),
+        Some(extension) => Err(Error::InvalidExtension(
+            extension.to_str().map(|s| s.to_string()),
```

Comment:
The `Option<String>` in the error can make for a very confusing error message. How about changing it to `String`, and using something like this here instead:
```suggestion
            extension.to_string_lossy().into_owned(),
```